saml-metadata-bindings/src/main/java/uk/gov/ida/saml/metadata/CertificateChainValidationFilter.java

```
@@ -64,22 +65,28 @@ public XMLObject filter(@Nullable XMLObject metadata) {
 
         try {
             if (metadata instanceof EntityDescriptor) {
-                processEntityDescriptor((EntityDescriptor) metadata);
+                if (processEntityDescriptor((EntityDescriptor) metadata) == EntityDescriptorFilteringResult.ROLE_DESCRIPTOR_LIST_EMPTY) {
+                    LOG.error("Fatal error - role descriptor list empty validating certificate chain, metadata will be filtered out");
+                    return null;
+                }
```

Might be worth making the pattern more like:

```
switch (thing) {
  case(EntityDescriptorFilteringResult.OK):
    break;
  case(EntityDescriptorFilteringResult.ROLE_DESCRIPTOR_LIST_EMPTY):
    LOG.error("...");
    return null;
  case():
  default:
     throw new SomeException("No case for " + thing);
}
```
In case someone adds a new error case to the enum and fails to update the handling code (which would currently get treated as an OK).

---

Really? It's basically a boolean.  I _really_ don't like having yet another "SomeException".  This feels like overkill for a very simple bit of code.

---

I guess it's less of a risk since these enums are private, but...

---

Maybe just flip it - `if (thing != OK) {handle error}` - at least it will never accidentally succeed.

---

Yep, that's fine too. As long as it fails closed, not open, 👍 from me.